### ISSUE TITLE
Allow regexp in id-blacklist

### ISSUE THREAD

**Issue by bensampaio** [eslint/eslint#10787]
**What rule do you want to change?**
id-blacklist

**Does this change cause the rule to produce more or fewer warnings?**
The same warnings I would say.

**How will the change be implemented? (New option, new default behavior, etc.)?**
Currently this rule receives a list of strings as options. With this change it would be possible to pass a list of strings and regular expressions. In the case of regular expressions they would be matched against identifiers and give a warning if they don't match.

**Please provide some example code that this change will affect:**

```js
const data = {};

const myCompany = () => {};

const myCompanyAccount = {};

const statusOfMyCompany = {};
```

**What does the rule currently do for this code?**
If the rule configuration is: `'ed-blacklist': ['error', 'data', 'myCompany']` then it only fails for the two first variables in the example above.

**What will the rule do after it's changed?**
If the rule configuration is: `'ed-blacklist': ['error', 'data', /myCompany/i]` then it would fail for all variables in the example above.

**Comment by kaicataldo** [comment]
This seems like a reasonable enhancement to me. Are you willing to make a PR if this is accepted?

As an aside, I believe this could technically also be done with [the id-match rule](https://eslint.org/docs/rules/id-match), but that doesn't seem scalable.

**Comment by bensampaio** [comment]
Hmm I see what you mean about [id-match rule](https://eslint.org/docs/rules/id-match). It would work but still I just want to blacklist some words not enforce how identifiers should be written.

I would like to contribute with a PR but I doubt I will have the time in the coming few months :(
